lib/matplotlib/backend_tools.py

```diff
@@ -622,28 +621,38 @@ def scroll_zoom(self, event):
         # https://gist.github.com/tacaswell/3144287
         if event.inaxes is None:
             return
-        ax = event.inaxes
-        cur_xlim = ax.get_xlim()
-        cur_ylim = ax.get_ylim()
-        # set the range
-        cur_xrange = (cur_xlim[1] - cur_xlim[0])*.5
-        cur_yrange = (cur_ylim[1] - cur_ylim[0])*.5
-        xdata = event.xdata  # get event x location
-        ydata = event.ydata  # get event y location
+
         if event.button == 'up':
             # deal with zoom in
-            scale_factor = 1 / self.base_scale
+            scl = 2.
         elif event.button == 'down':
             # deal with zoom out
-            scale_factor = self.base_scale
+            scl = 1/2.
         else:
             # deal with something that should never happen
-            scale_factor = 1
-        # set new limits
-        ax.set_xlim([xdata - cur_xrange*scale_factor,
-                     xdata + cur_xrange*scale_factor])
-        ax.set_ylim([ydata - cur_yrange*scale_factor,
-                     ydata + cur_yrange*scale_factor])
+            scl = 1
+
+        ax = event.inaxes
+        xmin, xmax, ymin, ymax = ax._get_view()
```

Comment:
The point of _get_view/_set_view is to hide the meaning of the "view" from outside of the Axes, because it is not necessarily Cartesian. The following calculation based on it should be placed in the Axes.